cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -792,13 +792,24 @@ private void monitorTwillApplication(final ScheduledExecutorService executor, fi
 
       // Monitor the application
       serviceController.set(controller);
+      final TwillController finalController = controller;
       controller.onTerminated(new Runnable() {
         @Override
         public void run() {
           if (executor.isShutdown()) {
             return;
           }
-          LOG.warn("{} was terminated; restarting with back-off", Constants.Service.MASTER_SERVICES);
+          Thread.interrupted();
```

Comment:
why interrupt?

Reply:
It is to clear the interrupt flag of this thread in case it is set, since we don't want the `Future.get()` below to throw interrupted exception.